File: java/Modulus calculator/MainActivity.java
package com.gradproject.moduluscalc;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    Button mod;
    EditText number1, number2;
    TextView result;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mod = findViewById(R.id.button1);
        number1 = findViewById(R.id.Entry1);
        number2 = findViewById(R.id.Entry2);
        result = findViewById(R.id.textView3);
    }
    public void modButton(View view){
        int myNumber1 = Integer.parseInt(number1.getText().toString());
            int myNumber2 = Integer.parseInt(number2.getText().toString());
            result.setText(""+(myNumber1 % myNumber2));


    }
}

File: Kotlin/age calculator/Complete app/app/src/main/java/com/badawy3/app1/MainActivity.kt
package com.badawy3.app1

import android.graphics.drawable.DrawableContainer
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button1.setOnClickListener({

            val yearofbirth: Int = EnterDOB.text.toString().toInt()
            val currentyear = Calendar.getInstance().get(Calendar.YEAR)
            val age = currentyear - yearofbirth
            textView.text = "your age is  " + age + " years"

        }
        )

    }
}